File: Python/signup.py
import uuid
import aiohttp
import asyncio
import os
from datetime import datetime
from dotenv import load_dotenv

# Load environment variables from .env file
load_dotenv()

# Import constants from config.py
from config import PORTAL_EX


async def signup():
    now = datetime.utcnow().isoformat()
    my_uuid = uuid.uuid4()
    username = f"python-example-{now}{my_uuid}"

    async with aiohttp.ClientSession() as session:
        async with session.post(
            f"{PORTAL_EX}/mobile/signup",
            json={"username": username, "isAccountAbstracted": False},
            headers={"Content-Type": "application/json"},
        ) as response:
            if response.status != 200:
                print(f"Failed to sign up: {await response.json()}")
                return
            data = await response.json()
            print(f"Successfully signed up user {username}!")
            client_api_key = data["clientApiKey"]

            # Write clientApiKey to a file
            with open("clientApiKey.txt", "w") as file:
                file.write(client_api_key)


if __name__ == "__main__":
    asyncio.run(signup())

File: Javascript/signup.js
const { v4: uuidv4 } = require('uuid');
const axios = require('axios');
const fs = require('fs');
const { PORTAL_EX } = require('./config');

async function Signup() {
  const now = new Date();
  const myUUID = uuidv4();
  const username = 'js-example-' + now.toISOString() + myUUID;
  const portalExResponse = await axios.post(
    `${PORTAL_EX}/mobile/signup`,
    { username: username, isAccountAbstracted: false },
    {
      headers: { 'Content-Type': 'application/json' },
    },
  );
  if (portalExResponse.status != 200) {
    console.error('Failed to sign up:', portalExResponse.data);
    return;
  }
  console.log(`Successfully signed up user ${username}!`);
  const clientApiKey = portalExResponse.data.clientApiKey;

  // Write clientApiKey to a file
  fs.writeFileSync('clientApiKey.txt', clientApiKey);
}

module.exports = Signup;
